### DISCUSSION TITLE
Configurable/Stylable Select Fields

### DISCUSSION THREAD

**Discussion by lukas-frey** [filamentphp/filament#4292] [Ideas]
Adding the option to style the results of the search items in a searchable Select Field.

I needed to customize the search field but still wanted to use Filament Forms so I added an option to configure how the search results look like.

And came up with a small solution that would let you use Blade views for the search result items.

The form field looked like this:
```
Search::make('record')
                ->preview()
                ->itemContent(view('livewire.templates.card'))
                ->choiceContent(view('livewire.templates.card'))
                ->disableLabel()
                ->searchable()
                ->searchPlaceholder('Zadejte název značky, kterou hledáte...')
                ->items(Manufacturer::query()->get()->mapWithKeys(fn($item, $key) => [$item->id => $item])->toArray())
                ->options(Manufacturer::query()->get()->pluck('name','id'))
                ->columns(3)
                ->required(),
```

Once again, it was just quickly thrown together and quite ugly. It'll need quite a bit of refactoring before being production ready. :)

The final search looks like this:

https://user-images.githubusercontent.com/10926334/193218709-1eb3f85e-9d86-40c4-8a01-a16b2ee3c9a2.mov

I'd like to hear your opinion and whether anyone would benefit from having this created and maybe merged into filament.

**Comment by danharrin** [comment]
Honestly our Select field is already causing so many issues that I don't want to make it any more complex. We already have a similar option - https://filamentphp.com/tricks/render-html-in-select-options

So if you want to release this, it would have to be in your own plugin

**Reply by lukas-frey** [reply]
> Honestly our Select field is already causing so many issues that I don't want to make it any more complex. We already have a similar option - https://filamentphp.com/tricks/render-html-in-select-options
> 
> So if you want to release this, it would have to be in your own plugin

I really need to check the tricks section more often, would save me a lot of headaches. :D For most use cases, the approach from the link is probably more than enough anyways. :)
